forge-game/src/main/java/forge/game/trigger/TriggerSpellAbilityCastOrCopy.java

```diff
@@ -306,10 +304,19 @@ public final void setTriggeringObjects(final SpellAbility sa, Map<AbilityKey, Ob
         sa.setTriggeringObject(AbilityKey.Card, castSA.getHostCard());
         sa.setTriggeringObject(AbilityKey.SpellAbility, castSA.copy(castSA.getHostCard(), true));
         sa.setTriggeringObject(AbilityKey.StackInstance, si);
-        if (!saForTargets.getTargets().isEmpty()) {
-            sa.setTriggeringObject(AbilityKey.SpellAbilityTarget, saForTargets.getTargets().get(0));
-        }
-        sa.setTriggeringObject(AbilityKey.SpellAbilityTargetingCards, saForTargets.getTargets().getTargetCards());
+        final List<TargetChoices> allTgts = saForTargets.getAllTargetChoices();
+        if (!allTgts.isEmpty()) {
+            boolean target = false;
+            final CardCollection targetedCards = new CardCollection();
```

Comment:
wouldn't a FCollection<GameObject> or GameEntity be better?

then we maybe could get rid of AbilityKey.SpellAbilityTarget

Reply:
Worth a try, let's see!

Reply:
Seems to be working!